sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -224,7 +226,16 @@ class ParquetFileFormat
             .orElse(filesByType.data.headOption)
             .toSeq
       }
-    ParquetFileFormat.mergeSchemasInParallel(filesToTouch, sparkSession)
+
+    filesToTouch match {
+      case f :: Nil =>
```

Comment:
if you are doing this, we should probably add an undocumented config option to control this. I also think the number can be bigger. We'd need to experiment with it though.